### PR TITLE
Backport of Migrate Benchmarks to 1.x

### PR DESCRIPTION
**Pull request by xabbu42**
Draft for now.
I need to recheck I've got everything important and everything still works.
@wipfli If you can also recheck I got everything we need from the benchmarks branch based on main that would be very helpful.

### CODE REVIEW

File: .github/workflows/upload-benchmarks.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
+name: Upload Benchmarks
+
+on:
+  push:
+    branches:
+      - main
+    tags: 
+      - v*
```
Comment (wipfli) [inline .github/workflows/upload-benchmarks.yml:8]: change to `v1*`

File: .github/workflows/upload-benchmarks.yml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+      - uses: actions/checkout@v2
+
+      - name: Use Node.js 14 x64
+        uses: actions/setup-node@v2
+        with:
+          node-version: 14
+          architecture: x64
+
+      - name: Install
+        run: npm ci
```
Comment (wipfli) [inline .github/workflows/upload-benchmarks.yml:23]: `yarn install --frozen-lockfile`

File: .github/workflows/upload-benchmarks.yml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+jobs:
+  upload_benchmarks:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Use Node.js 14 x64
+        uses: actions/setup-node@v2
+        with:
+          node-version: 14
```
Comment (wipfli) [inline .github/workflows/upload-benchmarks.yml:19]: node 10

File: .github/workflows/upload-benchmarks.yml [hunk 4]
```diff
@@ -0,0 +1,52 @@
+name: Upload Benchmarks
+
+on:
+  push:
+    branches:
+      - main
```
[the remark this answers is not in the document]
Comment (wipfli) [reply .github/workflows/upload-benchmarks.yml:6]: Even better remove this trigger

File: .github/workflows/upload-benchmarks.yml [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+          cp bench/versions/benchmarks_generated.js tmp
+          cp bench/versions/benchmarks_generated.js.map tmp
+
+      - name: Upload to GitHub Pages (main)
+        if: startsWith(github.ref, 'refs/heads/main')
+        uses: peaceiris/actions-gh-pages@v3
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          publish_dir: tmp
+          destination_dir: benchmarks/main
```
Comment (wipfli) [inline .github/workflows/upload-benchmarks.yml:40]: Remove

File: .github/workflows/upload-benchmarks.yml [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+        uses: actions/setup-node@v2
+        with:
+          node-version: 14
+          architecture: x64
+
+      - name: Install
+        run: npm ci
+
+      - name: Build benchmarks
+        run: npm run build-benchmarks
```
Comment (wipfli) [inline .github/workflows/upload-benchmarks.yml:26]: yarn

**Comment by wipfli** [comment]
Might want to also update the readme a little bit like in #443

**Comment by xabbu42** [comment]
I just ran the benchmarks and all of them still work, I believe this can be merged.

**Comment by xabbu42** [comment]
@wipfli Any chance this can be merged? Also a new 1.15 release would be useful. After that the default benchmark page in main should hopefully work without any compare parameter.

**Comment by HarelM** [comment]
I agree that these developers should be added to the dev team.
Seems like I don't have the right permissions... :-(
@nyurik @snickell Any chance to add these developers to the dev team?
Granting me access to add members can also do the trick :-)

**Comment by snickell** [comment]
Hi @HarelM , both you and @wipfli are marked as admins in the Manage Access console, (@wipfli directly, you transitively by membership in the core group)..... I don't know of a higher level of access, AFAIK Admin should permit adding to the dev group.... let me know if you have suggestions what I could change to enable your access, and any error messages or info from your side might help me out here! 🙏

**Pull request merged**